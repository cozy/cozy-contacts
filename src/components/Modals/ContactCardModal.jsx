import React from 'react'
import { PropTypes } from 'prop-types'
import { withContactsMutations } from '../../connections/allContacts'
import Modal, { ModalHeader, ModalContent } from 'cozy-ui/react/Modal'
import { Icon, Menu, MenuItem, Button } from 'cozy-ui/react'
import ContactCard from '../ContactCard/ContactCard'
import contactPropTypes from '../ContactPropTypes'

const ContactCardMenu = ({ deleteAction }) => (
  <Menu
    position="right"
    className="contact-card-modal__menu"
    component={
      <Button
        theme="secondary"
        extension="narrow"
        icon="dots"
        className="fix-c-btn"
      />
    }
  >
    <MenuItem
      className="menu__item--danger"
      icon={<Icon icon="delete" />}
      onSelect={deleteAction.action}
    >
      {deleteAction.label}
    </MenuItem>
  </Menu>
)
ContactCardMenu.propTypes = {
  deleteAction: PropTypes.shape({
    label: PropTypes.string.isRequired,
    action: PropTypes.func.isRequired
  }).isRequired
}

class ContactCardModal extends React.Component {
  state = {
    shouldDisplayConfirmDeleteModal: false
  }

  toggleConfirmDeleteModal = () => {
    this.setState(state => ({
      ...state,
      shouldDisplayConfirmDeleteModal: !state.shouldDisplayConfirmDeleteModal
    }))
  }

  deleteContact = async () => {
    const { contact, deleteContact, onDeleteContact } = this.props
    await deleteContact(contact)
    onDeleteContact(contact)
  }

  render() {
    const { onClose, contact } = this.props
    const { shouldDisplayConfirmDeleteModal } = this.state
    const { t } = this.context

    return (
      <Modal into="body" dismissAction={onClose} size="xlarge">
        <ContactCard
          title={t('contact_info')}
          contact={contact}
          renderHeader={children => (
            <ModalHeader className="contact-card-modal__header">
              {children}
              <ContactCardMenu
                deleteAction={{
                  label: t('delete'),
                  action: this.toggleConfirmDeleteModal
                }}
              />
            </ModalHeader>
          )}
          renderBody={children => <ModalContent>{children}</ModalContent>}
        />
        {shouldDisplayConfirmDeleteModal && (
          <Modal
            into="body"
            title={t('delete-confirmation.title')}
            description={t('delete-confirmation.description')}
            primaryText={t('delete')}
            primaryType="danger"
            primaryAction={this.deleteContact}
            secondaryText={t('cancel')}
            secondaryAction={this.toggleConfirmDeleteModal}
            dismissAction={this.toggleConfirmDeleteModal}
          />
        )}
      </Modal>
    )
  }
}

ContactCardModal.propTypes = {
  onClose: PropTypes.func.isRequired,
  contact: PropTypes.shape({
    name: contactPropTypes.name,
    phone: PropTypes.arrayOf(contactPropTypes.phone),
    email: PropTypes.arrayOf(contactPropTypes.email),
    address: PropTypes.arrayOf(contactPropTypes.address),
    cozy: PropTypes.arrayOf(contactPropTypes.cozy),
    birthday: contactPropTypes.birthday,
    note: contactPropTypes.note
  }).isRequired,
  deleteContact: PropTypes.func.isRequired,
  onDeleteContact: PropTypes.func.isRequired
}

export default withContactsMutations(ContactCardModal)
