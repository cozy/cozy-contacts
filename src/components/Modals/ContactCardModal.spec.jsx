import React from 'react'
import { DumpContactCardModal } from './ContactCardModal'
import { shallow, mount } from 'enzyme'
import AppLike from '../../tests/Applike'

const fakedContact = {
  id: '4e33fd27e1d8e55a34742bac6d16d3bd',
  _id: '4e33fd27e1d8e55a34742bac6d16d3bd',
  _type: 'io.cozy.contacts',
  _rev: '6-57f4ab78b1fb97bb90ea273ec881f196',
  address: [],
  company: '',
  cozy: [{ primary: true, url: 'https://q.mycozy.cloud' }],
  cozyMetadata: {
    createdAt: '2019-05-21T12:24:01.084Z',
    createdByApp: 'Contacts',
    createdByAppVersion: '0.8.3',
    doctypeVersion: 2,
    metadataVersion: 1,
    updatedAt: '2019-07-15T11:22:13.621Z',
    updatedByApps: [
      { date: '2019-07-15T11:22:13.621Z', slug: 'Contacts', version: '0.8.5' }
    ]
  },
  email: [{ address: 'quentin@cozycloud.cc', primary: true }],
  fullname: 'quentin valmori',
  me: true,
  metadata: { cozy: true, version: 1 },
  name: { familyName: 'valmori', givenName: 'quentin' },
  note: '',
  phone: [{ number: '+33600041300', primary: true }],
  relationships: {
    accounts: { data: [] },
    groups: {
      data: [
        {
          _id: '5b49553c5916cf7b5b2a5f48600078a8',
          _type: 'io.cozy.contacts.groups'
        }
      ]
    }
  },
  groups: {
    target: {
      id: '4e33fd27e1d8e55a34742bac6d16d3bd',
      _id: '4e33fd27e1d8e55a34742bac6d16d3bd',
      _type: 'io.cozy.contacts',
      _rev: '6-57f4ab78b1fb97bb90ea273ec881f196',
      company: '',
      fullname: 'quentin valmori',
      me: true,
      note: ''
    },
    name: 'groups',
    doctype: 'io.cozy.contacts.groups',
    data: [
      {
        _id: '5b49553c5916cf7b5b2a5f48600078a8',
        _type: 'io.cozy.contacts.groups'
      }
    ]
  },
  accounts: { name: 'accounts', doctype: 'io.cozy.contacts.accounts', data: [] }
}

const fakedAllGroups = [
  {
    id: '5b49553c5916cf7b5b2a5f48600078a8',
    _id: '5b49553c5916cf7b5b2a5f48600078a8',
    _type: 'io.cozy.contacts.groups',
    _rev: '1-7862def64fb044932d3264e2f8477454',
    cozyMetadata: {
      createdAt: '2019-07-15T11:22:13.551Z',
      createdByApp: 'Contacts',
      createdByAppVersion: '0.8.5',
      metadataVersion: 1,
      updatedAt: '2019-07-15T11:22:13.551Z',
      updatedByApps: [
        { date: '2019-07-15T11:22:13.551Z', slug: 'Contacts', version: '0.8.5' }
      ]
    },
    metadata: { version: 1 },
    name: '2018-2019 Enseignants'
  },
  {
    id: '8cb7ea7fe264260e997529439b0091c0',
    _id: '8cb7ea7fe264260e997529439b0091c0',
    _type: 'io.cozy.contacts.groups',
    _rev: '1-8b92242f1ccaca20e7862306cddbe37f',
    cozyMetadata: {
      createdAt: '2020-04-15T06:46:20.128Z',
      createdByApp: 'Contacts',
      createdByAppVersion: '0.8.7',
      metadataVersion: 1,
      updatedAt: '2020-04-15T06:46:20.128Z',
      updatedByApps: [
        { date: '2020-04-15T06:46:20.128Z', slug: 'Contacts', version: '0.8.7' }
      ]
    },
    metadata: { version: 1 },
    name: 'ez'
  },
  {
    id: '18c031b0ac9f47cdc48b275b1900726d',
    _id: '18c031b0ac9f47cdc48b275b1900726d',
    _type: 'io.cozy.contacts.groups',
    _rev: '1-f438477a749fa0b90cde2e7c53a41bd6',
    cozyMetadata: {
      createdAt: '2019-11-08T07:45:55.424Z',
      createdByApp: 'Contacts',
      createdByAppVersion: '0.8.6',
      metadataVersion: 1,
      updatedAt: '2019-11-08T07:45:55.424Z',
      updatedByApps: [
        { date: '2019-11-08T07:45:55.424Z', slug: 'Contacts', version: '0.8.6' }
      ]
    },
    metadata: { version: 1 },
    name: 'groupe test'
  }
]
const setup = ({
  fetchStatusContact,
  allGroupsContactStatus,
  editMode,
  shouldDisplayConfirmDeleteModal,
  contact,
  allGroups
}) => {
  return {
    fetchStatusContact,
    allGroupsContactStatus,
    editMode,
    shouldDisplayConfirmDeleteModal,
    contact,
    allGroups,
    t: x => x,
    toggleConfirmDeleteModal: jest.fn,
    toggleEditMode: jest.fn,
    deleteContact: jest.fn
  }
}

describe('ContactCardModal', () => {
  it('test if we display the Spinner', () => {
    const props = setup({
      fetchStatusContact: 'loading',
      allGroupsContactStatus: 'loading',
      editMode: false,
      shouldDisplayConfirmDeleteModal: false,
      contact: null,
      allGroups: null
    })
    const jsx = <DumpContactCardModal {...props} />
    const wrapper = shallow(jsx)
    expect(wrapper).toMatchSnapshot()
  })

  it('test if we display the contact card', () => {
    const props = setup({
      fetchStatusContact: 'loaded',
      allGroupsContactStatus: 'loaded',
      editMode: false,
      shouldDisplayConfirmDeleteModal: false,
      contact: fakedContact,
      allGroups: fakedAllGroups
    })
    const jsx = (
      <AppLike>
        <DumpContactCardModal {...props} />
      </AppLike>
    )
    const wrapper = mount(jsx)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('test if we display the delete confirmation modal', () => {
    const props = setup({
      fetchStatusContact: 'loaded',
      allGroupsContactStatus: 'loaded',
      editMode: false,
      shouldDisplayConfirmDeleteModal: true,
      contact: fakedContact,
      allGroups: fakedAllGroups
    })
    const jsx = (
      <AppLike>
        <DumpContactCardModal {...props} />
      </AppLike>
    )
    const wrapper = mount(jsx)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('test if we display the edit form modal', () => {
    const props = setup({
      fetchStatusContact: 'loaded',
      allGroupsContactStatus: 'loaded',
      editMode: true,
      shouldDisplayConfirmDeleteModal: false,
      contact: fakedContact,
      allGroups: fakedAllGroups
    })
    const jsx = (
      <AppLike>
        <DumpContactCardModal {...props} />
      </AppLike>
    )
    const wrapper = mount(jsx)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
