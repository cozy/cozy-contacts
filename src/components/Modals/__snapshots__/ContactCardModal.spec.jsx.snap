// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactCardModal test if we display the Spinner 1`] = `
<SpinnerContact
  size="xxlarge"
/>
`;

exports[`ContactCardModal test if we display the contact card 1`] = `"<div class=\\"styles__c-modal-header___hTCm5 u-flex u-flex-items-center u-flex-column-s u-pr-1-half-s\\"><div class=\\"u-flex u-flex-items-center u-flex-column-s u-flex-justify-start-s\\"><div class=\\"styles__c-avatar___PpDI- u-mb-half-s u-flex-shrink-0\\" style=\\"color: white;\\"><span class=\\"styles__c-avatar-initials___310qC\\">QV</span></div><h1 class=\\"u-title-h1 u-mv-0 u-mh-1 u-mb-half-s\\">quentin  valmori</h1></div><div class=\\"u-flex u-flex-row u-ml-auto u-ml-0-s\\"><div class=\\"u-flex-shrink-0 u-m-0\\"><div class=\\"css-1irp0at select--autowidth react-select__custom__container\\"><button type=\\"submit\\" class=\\"styles__c-btn___-2Vnj styles__c-btn--secondary___3Br_N styles__c-btn--small___3ArkI styles__c-btn--center___16_Xh\\"><span><svg class=\\"styles__icon___23x3R\\" style=\\"fill: var(--coolGrey);\\" width=\\"16\\" height=\\"16\\"><use xlink:href=\\"#icon-0\\"></use></svg><span>Manage groups</span><svg class=\\"group-manager__indicator styles__icon___23x3R\\" style=\\"fill: var(--coolGrey);\\" width=\\"12\\" height=\\"16\\"><use xlink:href=\\"#icon-0\\"></use></svg></span></button></div></div><div class=\\"styles__c-menu___2YTiU styles__c-menu--right___3wxbb u-mr-half\\"><button type=\\"submit\\" class=\\"styles__c-btn___-2Vnj styles__c-btn--secondary___3Br_N styles__c-btn--small___3ArkI styles__c-btn--narrow___27FHD styles__c-btn--center___16_Xh fix-c-btn\\" title=\\"Menu\\"><span><svg class=\\"styles__icon___23x3R\\" width=\\"12\\" height=\\"12\\" aria-hidden=\\"true\\" focusable=\\"false\\"><use xlink:href=\\"#dots\\"></use></svg><span class=\\"u-visuallyhidden\\">Menu</span></span></button></div></div></div><div class=\\"styles__c-modal-content___2av1P\\"><div class=\\"u-mb-2\\"><h3 class=\\"u-title-h2 u-mt-1-half u-mb-1\\">Groups</h3><ol class=\\"u-nolist u-m-0 u-p-0\\"><li class=\\"u-dib u-slateGrey u-fz-small u-p-half u-mr-half u-w-auto u-maw-6 u-bg-paleGrey u-ellipsis\\">2018-2019 Enseignants</li></ol></div><div><h3 class=\\"u-title-h2 u-mt-1-half u-mb-1\\">Contact informations</h3><ol class=\\"u-nolist u-m-0 u-p-0 u-pl-1-half u-pl-0-s\\"><li><div class=\\"u-flex u-mt-half\\"><div class=\\"u-mr-1 u-fz-large\\"><svg class=\\"styles__icon___23x3R\\" style=\\"fill: var(--coolGrey);\\" width=\\"16\\" height=\\"16\\"><use xlink:href=\\"#icon-0\\"></use></svg></div><div><div class=\\"u-mb-half u-breakword u-fz-medium\\"><a href=\\"tel:+33600041300\\" class=\\"u-link\\">+33600041300</a></div></div></div></li><li><div class=\\"u-flex u-mt-half\\"><div class=\\"u-mr-1 u-fz-large\\"><svg class=\\"styles__icon___23x3R\\" style=\\"fill: var(--coolGrey);\\" width=\\"16\\" height=\\"16\\"><use xlink:href=\\"#icon-0\\"></use></svg></div><div><div class=\\"u-mb-half u-breakword u-fz-medium\\"><a href=\\"mailto:quentin@cozycloud.cc\\" class=\\"u-link\\">quentin@cozycloud.cc</a></div></div></div></li><li><div class=\\"u-flex u-mt-half\\"><div class=\\"u-mr-1 u-fz-large\\"><svg class=\\"styles__icon___23x3R\\" style=\\"fill: var(--coolGrey);\\" width=\\"16\\" height=\\"16\\"><use xlink:href=\\"#icon-0\\"></use></svg></div><div><div class=\\"u-mb-half u-breakword u-fz-medium\\"><a href=\\"https://q.mycozy.cloud\\" class=\\"u-link\\">https://q.mycozy.cloud</a></div></div></div></li></ol></div></div>"`;

exports[`ContactCardModal test if we display the delete confirmation modal 1`] = `"<div class=\\"styles__c-modal-container___31zmi\\"><div class=\\"styles__c-overlay___31iWd\\"><div class=\\"styles__c-modal-wrapper___8PyE9\\"><div></div><div class=\\"styles__c-modal___33aV9 styles__c-modal--small___XUfOM styles__c-modal--closable___29CLQ\\" role=\\"dialog\\" aria-modal=\\"true\\" aria-labelledby=\\"modal_1\\"><button class=\\"styles__c-modal-close___1v2bp\\" type=\\"button\\" aria-label=\\"close\\"><svg class=\\"styles__icon___23x3R\\" style=\\"fill: var(--coolGrey);\\" width=\\"24\\" height=\\"24\\"><use xlink:href=\\"#cross\\"></use></svg></button><div class=\\"styles__c-modal-header___hTCm5\\" id=\\"modal_1\\"><h2>delete-confirmation.title</h2></div><div class=\\"styles__c-modal-content___2av1P\\">delete-confirmation.description-simple</div><div class=\\"styles__c-modal-footer___2Z4Tc\\"><div class=\\"u-flex u-flex-wrap-reverse u-flex-justify-end styles__c-modal-footer--button___1sIa4\\"><button type=\\"submit\\" class=\\"styles__c-btn___-2Vnj styles__c-btn--secondary___3Br_N styles__c-btn--center___16_Xh u-flex-auto-s\\"><span><span>cancel</span></span></button><button type=\\"submit\\" class=\\"styles__c-btn___-2Vnj styles__c-btn--danger___17T_C styles__c-btn--center___16_Xh u-flex-auto-s\\"><span><span>delete</span></span></button></div></div></div></div></div></div>"`;

exports[`ContactCardModal test if we display the edit form modal 1`] = `"<div class=\\"styles__c-modal-container___31zmi\\"><div class=\\"styles__c-overlay___31iWd\\"><div class=\\"styles__c-modal-wrapper___8PyE9 styles__c-modal-wrapper--fullscreen___3ygpX\\"><div></div><div class=\\"styles__c-modal___33aV9 styles__c-modal--xlarge___1XePQ styles__c-modal--overflowHidden___3H87t styles__c-modal--fullscreen___8xZVw styles__c-modal--closable___29CLQ\\" role=\\"dialog\\" aria-modal=\\"true\\"><button class=\\"styles__c-modal-close___1v2bp styles__c-modal-close--notitle___1rqne\\" type=\\"button\\" aria-label=\\"close\\"><svg class=\\"styles__icon___23x3R\\" style=\\"fill: var(--coolGrey);\\" width=\\"24\\" height=\\"24\\"><use xlink:href=\\"#cross\\"></use></svg></button><div class=\\"styles__c-modal-header___hTCm5\\"><h2>edit-contact</h2></div><div class=\\"styles__c-modal-content___2av1P\\"><div><form class=\\"u-flex u-flex-column u-flex-items-stretch u-flex-justify-start\\"><h3 class=\\"u-title-h2 u-mt-1-half u-mb-1\\">Contact informations</h3><div class=\\"contact-form-field\\"><label for=\\"field_3\\" class=\\"styles__c-label___o4ozG contact-form__label u-inline-flex u-flex-items-center u-mr-half\\"><svg class=\\"u-mr-1 styles__icon___23x3R\\" style=\\"fill: var(--coolGrey);\\" width=\\"16\\" height=\\"16\\"><use xlink:href=\\"#people\\"></use></svg><span class=\\"u-w-4\\">Firstname</span></label><div class=\\"contact-form__inputs-wrapper\\"><div class=\\"contact-form__input-wrapper\\"><div class=\\"contact-form__input-container\\"><input type=\\"text\\" id=\\"field_3\\" class=\\"styles__c-input-text___3TAv1 styles__c-input-text--medium___28jPV styles__c-input-text--fullwidth___33o_f contact-form__input\\" placeholder=\\"\\" name=\\"givenName\\" data-valid=\\"true\\" value=\\"quentin\\"></div></div></div></div><div class=\\"contact-form-field\\"><label for=\\"field_4\\" class=\\"styles__c-label___o4ozG contact-form__label u-inline-flex u-flex-items-center u-mr-half u-pl-0-s u-pl-2\\"><span class=\\"u-w-4\\">Lastname</span></label><div class=\\"contact-form__inputs-wrapper\\"><div class=\\"contact-form__input-wrapper\\"><div class=\\"contact-form__input-container\\"><input type=\\"text\\" id=\\"field_4\\" class=\\"styles__c-input-text___3TAv1 styles__c-input-text--medium___28jPV styles__c-input-text--fullwidth___33o_f contact-form__input\\" placeholder=\\"\\" name=\\"familyName\\" required=\\"\\" data-valid=\\"true\\" value=\\"valmori\\"></div></div></div></div><div class=\\"contact-form-field\\"><label for=\\"field_5\\" class=\\"styles__c-label___o4ozG contact-form__label u-inline-flex u-flex-items-center u-mr-half\\"><svg class=\\"u-mr-1 styles__icon___23x3R\\" style=\\"fill: var(--coolGrey);\\" width=\\"16\\" height=\\"16\\"><use xlink:href=\\"#icon-0\\"></use></svg><span class=\\"u-w-4\\">Phone</span></label><div class=\\"contact-form__inputs-wrapper\\"><div class=\\"contact-form__inputs-line\\"><div class=\\"contact-form__input-wrapper\\"><div class=\\"contact-form__input-container\\"><input type=\\"tel\\" id=\\"field_5\\" class=\\"styles__c-input-text___3TAv1 styles__c-input-text--medium___28jPV styles__c-input-text--fullwidth___33o_f contact-form__input\\" placeholder=\\"\\" name=\\"phone[0].phone\\" data-valid=\\"true\\" value=\\"+33600041300\\"></div><div class=\\"contact-form__input-container u-mt-half-s u-pl-half u-pl-0-s\\"><input type=\\"text\\" class=\\"styles__c-input-text___3TAv1 styles__c-input-text--medium___28jPV styles__c-input-text--fullwidth___33o_f contact-form__input\\" placeholder=\\"Label\\" name=\\"phone[0].phoneLabel\\" data-valid=\\"true\\" value=\\"\\"></div></div><button type=\\"button\\" class=\\"styles__c-btn___-2Vnj styles__c-btn--secondary___3Br_N styles__c-btn--center___16_Xh styles__c-btn--round___1Lkyl contact-form__meta-button u-m-0\\" title=\\"delete\\"><span><svg class=\\"styles__icon___23x3R\\" width=\\"16\\" height=\\"16\\" aria-hidden=\\"true\\" focusable=\\"false\\"><use xlink:href=\\"#cross-small\\"></use></svg><span class=\\"u-visuallyhidden\\">delete</span></span></button></div><button type=\\"button\\" class=\\"styles__c-btn___-2Vnj styles__c-btn--text___2Vp-2 styles__c-btn--center___16_Xh u-ph-0 u-mh-0\\"><span><svg class=\\"styles__icon___23x3R\\" width=\\"16\\" height=\\"16\\" aria-hidden=\\"true\\" focusable=\\"false\\"><use xlink:href=\\"#plus\\"></use></svg><span>Add a phone</span></span></button></div></div><div class=\\"contact-form-field\\"><label for=\\"field_6\\" class=\\"styles__c-label___o4ozG contact-form__label u-inline-flex u-flex-items-center u-mr-half\\"><svg class=\\"u-mr-1 styles__icon___23x3R\\" style=\\"fill: var(--coolGrey);\\" width=\\"16\\" height=\\"16\\"><use xlink:href=\\"#icon-0\\"></use></svg><span class=\\"u-w-4\\">Email</span></label><div class=\\"contact-form__inputs-wrapper\\"><div class=\\"contact-form__inputs-line\\"><div class=\\"contact-form__input-wrapper\\"><div class=\\"contact-form__input-container\\"><input type=\\"email\\" id=\\"field_6\\" class=\\"styles__c-input-text___3TAv1 styles__c-input-text--medium___28jPV styles__c-input-text--fullwidth___33o_f contact-form__input\\" placeholder=\\"\\" name=\\"email[0].email\\" data-valid=\\"true\\" value=\\"quentin@cozycloud.cc\\"></div><div class=\\"contact-form__input-container u-mt-half-s u-pl-half u-pl-0-s\\"><input type=\\"text\\" class=\\"styles__c-input-text___3TAv1 styles__c-input-text--medium___28jPV styles__c-input-text--fullwidth___33o_f contact-form__input\\" placeholder=\\"Label\\" name=\\"email[0].emailLabel\\" data-valid=\\"true\\" value=\\"\\"></div></div><button type=\\"button\\" class=\\"styles__c-btn___-2Vnj styles__c-btn--secondary___3Br_N styles__c-btn--center___16_Xh styles__c-btn--round___1Lkyl contact-form__meta-button u-m-0\\" title=\\"delete\\"><span><svg class=\\"styles__icon___23x3R\\" width=\\"16\\" height=\\"16\\" aria-hidden=\\"true\\" focusable=\\"false\\"><use xlink:href=\\"#cross-small\\"></use></svg><span class=\\"u-visuallyhidden\\">delete</span></span></button></div><button type=\\"button\\" class=\\"styles__c-btn___-2Vnj styles__c-btn--text___2Vp-2 styles__c-btn--center___16_Xh u-ph-0 u-mh-0\\"><span><svg class=\\"styles__icon___23x3R\\" width=\\"16\\" height=\\"16\\" aria-hidden=\\"true\\" focusable=\\"false\\"><use xlink:href=\\"#plus\\"></use></svg><span>Add an email</span></span></button></div></div><div class=\\"contact-form-field\\"><label for=\\"field_7\\" class=\\"styles__c-label___o4ozG contact-form__label u-inline-flex u-flex-items-center u-mr-half\\"><svg class=\\"u-mr-1 styles__icon___23x3R\\" style=\\"fill: var(--coolGrey);\\" width=\\"16\\" height=\\"16\\"><use xlink:href=\\"#icon-0\\"></use></svg><span class=\\"u-w-4\\">Address</span></label><div class=\\"contact-form__inputs-wrapper\\"><div class=\\"contact-form__inputs-line\\"><div class=\\"contact-form__input-wrapper\\"><div class=\\"contact-form__input-container\\"><input type=\\"text\\" id=\\"field_7\\" class=\\"styles__c-input-text___3TAv1 styles__c-input-text--medium___28jPV styles__c-input-text--fullwidth___33o_f contact-form__input\\" placeholder=\\"\\" name=\\"address[0].address\\" data-valid=\\"true\\" value=\\"\\"></div></div></div><button type=\\"button\\" class=\\"styles__c-btn___-2Vnj styles__c-btn--text___2Vp-2 styles__c-btn--center___16_Xh u-ph-0 u-mh-0\\"><span><svg class=\\"styles__icon___23x3R\\" width=\\"16\\" height=\\"16\\" aria-hidden=\\"true\\" focusable=\\"false\\"><use xlink:href=\\"#plus\\"></use></svg><span>Add an address</span></span></button></div></div><div class=\\"contact-form-field\\"><label for=\\"field_8\\" class=\\"styles__c-label___o4ozG contact-form__label u-inline-flex u-flex-items-center u-mr-half\\"><svg class=\\"u-mr-1 styles__icon___23x3R\\" style=\\"fill: var(--coolGrey);\\" width=\\"16\\" height=\\"16\\"><use xlink:href=\\"#icon-0\\"></use></svg><span class=\\"u-w-4\\">Cozy URL</span></label><div class=\\"contact-form__inputs-wrapper\\"><div class=\\"contact-form__input-wrapper\\"><div class=\\"contact-form__input-container\\"><input type=\\"url\\" id=\\"field_8\\" class=\\"styles__c-input-text___3TAv1 styles__c-input-text--medium___28jPV styles__c-input-text--fullwidth___33o_f contact-form__input\\" placeholder=\\"\\" name=\\"cozy\\" data-valid=\\"true\\" value=\\"https://q.mycozy.cloud\\"></div><div class=\\"contact-form__input-container u-mt-half-s u-pl-half u-pl-0-s\\"><input type=\\"text\\" class=\\"styles__c-input-text___3TAv1 styles__c-input-text--medium___28jPV styles__c-input-text--fullwidth___33o_f contact-form__input\\" placeholder=\\"Label\\" name=\\"cozyLabel\\" data-valid=\\"true\\" value=\\"\\"></div></div></div></div><div class=\\"contact-form-field\\"><label for=\\"field_9\\" class=\\"styles__c-label___o4ozG contact-form__label u-inline-flex u-flex-items-center u-mr-half\\"><svg class=\\"u-mr-1 styles__icon___23x3R\\" style=\\"fill: var(--coolGrey);\\" width=\\"16\\" height=\\"16\\"><use xlink:href=\\"#icon-0\\"></use></svg><span class=\\"u-w-4\\">Company</span></label><div class=\\"contact-form__inputs-wrapper\\"><div class=\\"contact-form__input-wrapper\\"><div class=\\"contact-form__input-container\\"><input type=\\"text\\" id=\\"field_9\\" class=\\"styles__c-input-text___3TAv1 styles__c-input-text--medium___28jPV styles__c-input-text--fullwidth___33o_f contact-form__input\\" placeholder=\\"\\" name=\\"company\\" data-valid=\\"true\\" value=\\"\\"></div></div></div></div><div class=\\"contact-form-field\\"><label for=\\"field_10\\" class=\\"styles__c-label___o4ozG contact-form__label u-inline-flex u-flex-items-center u-mr-half\\"><svg class=\\"u-mr-1 styles__icon___23x3R\\" style=\\"fill: var(--coolGrey);\\" width=\\"16\\" height=\\"16\\"><use xlink:href=\\"#icon-0\\"></use></svg><span class=\\"u-w-4\\">Birthday</span></label><div class=\\"contact-form__inputs-wrapper\\"><div class=\\"contact-form__input-wrapper\\"><div class=\\"contact-form__input-container\\"><input type=\\"date\\" id=\\"field_10\\" class=\\"styles__c-input-text___3TAv1 styles__c-input-text--medium___28jPV styles__c-input-text--fullwidth___33o_f contact-form__input\\" placeholder=\\"\\" name=\\"birthday\\" data-valid=\\"true\\" value=\\"\\"></div></div></div></div><div class=\\"contact-form-field\\"><label for=\\"field_11\\" class=\\"styles__c-label___o4ozG contact-form__label u-inline-flex u-flex-items-center u-mr-half\\"><svg class=\\"u-mr-1 styles__icon___23x3R\\" style=\\"fill: var(--coolGrey);\\" width=\\"16\\" height=\\"16\\"><use xlink:href=\\"#icon-0\\"></use></svg><span class=\\"u-w-4\\">Notes</span></label><div class=\\"contact-form__inputs-wrapper\\"><div class=\\"contact-form__input-wrapper\\"><div class=\\"contact-form__input-container\\"><textarea placeholder=\\"\\" class=\\"styles__c-textarea___D7EEH styles__c-textarea--medium___T8f3b styles__c-textarea--fullwidth___Ih_mg contact-form__input\\" name=\\"note\\" id=\\"field_11\\" type=\\"textarea\\" data-valid=\\"true\\"></textarea></div></div></div></div></form></div></div><div class=\\"styles__c-modal-footer___2Z4Tc u-ta-right\\"><button type=\\"button\\" class=\\"styles__c-btn___-2Vnj styles__c-btn--secondary___3Br_N styles__c-btn--center___16_Xh\\"><span><span>Cancel</span></span></button><button type=\\"submit\\" class=\\"styles__c-btn___-2Vnj styles__c-btn--center___16_Xh\\"><span><span>Save</span></span></button></div></div></div></div></div>"`;
